p2p/protocol/identify/id.go

```diff
@@ -316,6 +317,16 @@ func (ids *IDService) populateMessage(mes *pb.Identify, c network.Conn) {
 	}
 	log.Debugf("%s sent listen addrs to %s: %s", c.LocalPeer(), c.RemotePeer(), laddrs)
 
+	// Generate a signed routing record containing our listen addresses and
+	// send it along with the unsigned addrs
+	signedState, err := host.SignedRoutingStateFromHost(ids.Host)
+	if err != nil {
+		envelopeBytes, err := signedState.Marshal()
+		if err != nil {
+			mes.SignedRoutingState = envelopeBytes
+		}
+	}
```

Comment:
let's not swallow the errors here, warn on them?

Reply:
good call. I totally had the if condition backwards too, lol.